modules/labs-react/search-form/lib/SearchForm.tsx

```diff
@@ -118,173 +117,318 @@ function getInputColors(theme: SearchThemeAttributes, isFocused?: boolean) {
+      '&::-webkit-search-cancel-button': {
+        display: 'none',
+      },
+      '&::placeholder': {
+        color: searchInputPlaceholderColor,
       },
+      '&:placeholder-shown': {
+        textOverflow: 'ellipsis',
+      },
+
       '&:hover': {
-        backgroundColor: inputColors.backgroundHover,
+        backgroundColor: searchInputBackgroundHover,
+      },
+
+      '&:is(:focus-visible, &.focus):where(:not([disabled]))': {
+        background: searchInputBackgroundFocus,
+        color: searchInputColorFocus,
+        borderColor: brand.common.focusOutline,
+        outline: `${px2rem(2)} solid transparent`,
+        boxShadow: cssVar(
+          searchInputBoxShadowFocus,
+          `inset 0 0 0 ${px2rem(1)} ${brand.common.focusOutline}`
+        ),
+        '::placeholder': {
+          color: searchInputPlaceholderColorFocus,
+        },
       },
     },
-    ...collapseStyles,
-  };
+  }),
+  modifiers: {
+    isHiddenSubmitSearchIcon: {
+      true: ({submitSearchIconPart}) => ({
+        [submitSearchIconPart]: {
+          display: 'none',
+        },
+      }),
+    },
+    isHiddenOpenSeachIcon: {
+      true: ({openSearchIconPart}) => ({
+        [openSearchIconPart]: {
+          display: 'none',
+        },
+      }),
+    },
+    isCollapsed: {
+      true: ({
+        searchFieldPart,
+        submitSearchIconPart,
+        openSearchIconPart,
+        searchInputPart,
+        height,
+      }) => ({
+        top: 0,
+        right: 0,
+        left: 0,
+        bottom: 0,
+        margin: 0,
+        position: 'relative',
+        backgroundColor: system.color.bg.transparent,
+        transition: 'background-color 120ms',
+        maxWidth: calc.add(system.space.x10, system.space.x2),
+        minWidth: calc.add(system.space.x10, system.space.x3),
+        overflow: 'hidden',
+        zIndex: 1,
+        [searchFieldPart]: {
+          display: 'none',
+          maxWidth: '100%',
+        },
+        [`${submitSearchIconPart}, ${openSearchIconPart}`]: {
+          margin: `auto ${system.space.x2}`,
+        },
+        [searchInputPart]: {
+          fontSize: '20px',
+          paddingInlineStart: calc.add(system.space.x10, system.space.x4),
+          paddingInlineEnd: calc.add(system.space.x10, system.space.x4),
+          maxWidth: 'none',
+          minWidth: 0,
+          backgroundColor: system.color.bg.transparent,
+          height,
+        },
+      }),
+      false: ({searchFieldPart}) => ({
+        [searchFieldPart]: {
+          display: 'inline-block',
+        },
+      }),
+    },
+    rightAlign: {
+      true: {
+        textAlign: 'right',
+      },
+    },
+    showForm: {
+      true: {},
+      false: {},
+    },
+    grow: {
+      true: {
+        maxWidth: '100%',
+        [`[data-part="search-form-field"], [data-part="search-form-input"]`]: {
+          maxWidth: '100%',
+        },
+      },
+    },
+    searchTheme: {
+      // Light theme
+      0: ({searchInputPart}) => ({
```

Comment:
I'd rather change the enum and make this be `'light' | 'dark' | 'transparent'`. We should also update the `enum` to be an object instead. There shouldn't be a breaking change if people use `SearchTheme.Light`. I doubt anybody is passing `0`: `<SearchForm searchTheme={0}>`

Reply:
ok good, it is a labs component so that might better